packages/framework/attributor/src/runtimeAttributorDataStoreChannel.ts

```diff
@@ -0,0 +1,354 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { bufferToString, TypedEventEmitter } from "@fluid-internal/client-utils";
+import { AttachState, IDeltaManager } from "@fluidframework/container-definitions/internal";
+import {
+	FluidObject,
+	IFluidHandle,
+	IRequest,
+	IResponse,
+} from "@fluidframework/core-interfaces";
+import type { IFluidHandleInternal } from "@fluidframework/core-interfaces/internal";
+import { assert, Deferred, unreachableCase } from "@fluidframework/core-utils/internal";
+import { FluidObjectHandle } from "@fluidframework/datastore/internal";
+import { IFluidDataStoreRuntimeEvents } from "@fluidframework/datastore-definitions/internal";
+import {
+	IDocumentMessage,
+	type ISnapshotTree,
+	ISequencedDocumentMessage,
+	IQuorumClients,
+} from "@fluidframework/driver-definitions/internal";
+import {
+	IGarbageCollectionData,
+	IFluidDataStoreChannel,
+	IFluidDataStoreContext,
+	IInboundSignalMessage,
+	VisibilityState,
+	type ISummaryTreeWithStats,
+	type ITelemetryContext,
+	type AttributionInfo,
+	type AttributionKey,
+} from "@fluidframework/runtime-definitions/internal";
+import { SummaryTreeBuilder } from "@fluidframework/runtime-utils/internal";
+import {
+	ITelemetryLoggerExt,
+	MonitoringContext,
+	raiseConnectedEvent,
+	createChildMonitoringContext,
+} from "@fluidframework/telemetry-utils/internal";
+
+import { OpStreamAttributor, type IAttributor } from "./attributor.js";
+import { opBlobName, type IRuntimeAttributor } from "./attributorContracts.js";
+import { AttributorSerializer, chain, deltaEncoder, type Encoder } from "./encoders.js";
+import { makeLZ4Encoder } from "./lz4Encoder.js";
+
+/**
+ * Data store channel for the runtime attributor. This channel is responsible for storing and managing the
+ */
+export class RuntimeAttributorDataStoreChannel
+	extends TypedEventEmitter<IFluidDataStoreRuntimeEvents>
+	implements IFluidDataStoreChannel, IRuntimeAttributor
+{
+	public constructor(
+		public readonly dataStoreContext: IFluidDataStoreContext,
+		existing: boolean,
+	) {
+		super();
+		this.mc = createChildMonitoringContext({
+			logger: dataStoreContext.baseLogger,
+			namespace: "Attributor",
+		});
+		this._attachState = dataStoreContext.attachState;
+		if (existing) {
+			this.visibilityState =
+				dataStoreContext.attachState === AttachState.Detached
+					? VisibilityState.LocallyVisible
+					: VisibilityState.GloballyVisible;
+		} else {
+			this.visibilityState = VisibilityState.NotVisible;
+		}
+		// If it's existing we know it has been attached.
+		if (existing) {
+			this.deferredAttached.resolve();
+		}
+		this.entryPoint = new FluidObjectHandle<FluidObject>(
```

Comment:
Something to think about - Would it be better to separate the implementation of `IRuntimeAttributor` out from the implementation of `IFluidDataStoreChannel`?  Could we even reuse `FluidDataStoreRuntime` class?  It takes in a callback `provideEntryPoint`, you can have that return your `IRuntimeAttributor`.

Reply:
If it works to use `FluidDataStoreRuntime` class I would strongly be in favor of doing that now.

Reply:
Hmm I guess maybe this isn't the way to go now that I think more about it - Unless you really can just use DDSes to store the data, then it seems pretty clean.  You might even find yourself wanting to build on Aqueduct...

I think it's fine to start out fully custom, but as you add more capabilities coming up, consider how "special" this thing really is, v. falling into established patterns where you can write way less code

Reply:
In the meantime, do consider at least having 2 different classes, tied together via `entryPoint`, rather than having `this,entryPoint` be set to `this`.  As it is, the fields and methods of the two different interfaces are all mingled together and it makes both less clear.

Reply:
I separated the 2 classes, but I don't think I can use the default FluidDataStoreRuntime class as it mostly does not allow me to override certain methods, and it also differs in ways like allowing creation of ddses etc.